e2e/basic/reverse_proxy_test.go

```diff
@@ -0,0 +1,55 @@
+package basic
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/knuu/pkg/knuu"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReverseProxy(t *testing.T) {
+	t.Parallel()
+	// Setup
+
+	main, err := knuu.NewInstance("main")
+	require.NoError(t, err, "Error creating instance")
+
+	err = main.SetImage("alpine:latest")
+	require.NoError(t, err, "Error setting image")
+
+	err = main.SetCommand("sleep", "infinite")
+	require.NoError(t, err, "Error executing command")
+
+	require.NoError(t, main.Commit(), "Error committing instance")
+
+	t.Cleanup(func() {
+		if os.Getenv("KNUU_SKIP_CLEANUP") == "true" {
+			t.Log("Skipping cleanup")
+			return
+		}
+
+		require.NoError(t, main.Destroy(), "Error destroying instance")
+	})
+
+	require.NoError(t, main.EnableBitTwister(), "Error enabling BitTwister")
+	require.NoError(t, main.Start(), "Error starting main instance")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	require.NoError(t, main.BitTwister.WaitForStart(ctx), "Error waiting for BitTwister to start")
+
+	// test if BitTwister running in a sidecar is accessible
+	err = main.SetBandwidthLimit(1000)
+	assert.NoError(t, err, "Error setting bandwidth limit")
+
+	// Check if the BitTwister service is set
+	out, err := main.BitTwister.Client().AllServicesStatus()
+	require.NoError(t, err, "Error getting all services status")
+	assert.NotEmpty(t, out, "No services found")
```

Comment:
If possible it would be good to check for some explicit state other than not empty as this would not protect us from a bug in the code that returns an invalid non empty response.

Reply:
we could also use an nginx server instead

Reply:
delicate point, updated

Reply:
What was the resolution here?

Reply:
https://github.com/celestiaorg/knuu/pull/336/files#diff-c6b5c20a40f4e36e513597308a06b174baa20c53eea0fe97fd057dadb9821ddcR56-R58

Reply:
instead of just checking if it is empty or not, now we check for existence of a particular field which can yield only if the request goes through successfully.